test/unix/test_lwt_bytes.ml

```diff
@@ -496,4 +497,209 @@ let suite = suite "lwt_bytes" [
       let check = "abcaaa" = Lwt_bytes.to_string buf in
       Lwt.return check
     end;
+
+    test "bytes read" begin fun () ->
+      let test_file = "bytes_io_data" in
+      Lwt_unix.openfile test_file [O_RDONLY] 0
+      >>= fun fd ->
+      let buf = Lwt_bytes.create 6 in
+      Lwt_bytes.read fd buf 0 6
+      >>= fun _n ->
+      let check = "abcdef" = Lwt_bytes.to_string buf in
+      Lwt_unix.close fd
+      >>= fun () ->
+      Lwt.return check
+    end;
+
+    test "bytes write" begin fun () ->
+      let test_file = "bytes_io_data_write" in
+      Lwt_unix.openfile test_file [O_RDWR;O_TRUNC; O_CREAT] 0o666
+      >>= fun fd ->
+      let buf_write = Lwt_bytes.of_string "abc" in
+      Lwt_bytes.write fd buf_write 0 3
+      >>= fun _n ->
+      Lwt_unix.close fd
+      >>= fun () ->
+      Lwt_unix.openfile test_file [O_RDONLY] 0
+      >>= fun fd ->
+      let buf_read = Lwt_bytes.create 3 in
+      Lwt_bytes.read fd buf_read 0 3
+      >>= fun _n ->
+      let check = buf_write = buf_read in
+      Lwt_unix.close fd
+      >>= fun () ->
+      Lwt.return check
+    end;
+
+    test "bytes recv" ~only_if:(fun () -> not Sys.win32) begin fun () ->
+      let server_is_ready, notify_server_is_ready = Lwt.wait () in
+      let buf = Lwt_bytes.create 6 in
+      let server () =
+        let sock = Lwt_unix.socket Lwt_unix.PF_INET Lwt_unix.SOCK_STREAM 0 in
+        let sockaddr = Lwt_unix.ADDR_INET (Unix.inet_addr_loopback, 0) in
+        Lwt_unix.bind sock sockaddr
+        >>= fun () ->
+        let server_address = Lwt_unix.getsockname sock in
+        Lwt.wakeup_later notify_server_is_ready server_address;
```

Comment:
IIRC this should be moved under `listen`.

Reply:
You mean something like this ?:

```diff
diff --git a/test/unix/test_lwt_bytes.ml b/test/unix/test_lwt_bytes.ml
index 37d8a0c0..8ffca4ff 100644
--- a/test/unix/test_lwt_bytes.ml
+++ b/test/unix/test_lwt_bytes.ml
@@ -540,8 +540,8 @@ let suite = suite "lwt_bytes" [
         Lwt_unix.bind sock sockaddr
         >>= fun () ->
         let server_address = Lwt_unix.getsockname sock in
-        Lwt.wakeup_later notify_server_is_ready server_address;
         let () = Lwt_unix.listen sock 5 in
+        Lwt.wakeup_later notify_server_is_ready server_address;
         Lwt_unix.accept sock
         >>= fun (fd_client, _) ->
         Lwt_unix.write_string fd_client "abcdefghij" 0 9
```

Reply:
Yes ;)